plugins/woocommerce/includes/admin/class-wc-admin-notices.php

```diff
@@ -115,6 +119,46 @@ public static function reset_admin_notices() {
 		self::add_notice( 'template_files' );
 		self::add_min_version_notice();
 		self::add_maxmind_missing_license_key_notice();
+		self::maybe_add_legacy_api_removal_notice();
+	}
+
+	/**
+	 * Add an admin notice about the removal of the Legacy REST API if the said API is enabled.
+	 *
+	 * TODO: Remove this method in WooCommerce 9.0.
+	 */
+	private static function maybe_add_legacy_api_removal_notice() {
+		if ( 'yes' !== get_option( 'woocommerce_api_enabled' ) ) {
+			return;
+		}
+
```

Comment:
![double-notice](https://github.com/woocommerce/woocommerce/assets/916023/8c44e136-935f-4182-8645-9b6f055c138c)

Would it be worth adding some logic here to not show the notice on the actual Legacy API screen, since the same notice is shown elsewhere on the page?

Reply:
I don't think it's worth the effort. It's a temporary change, this is not a settings page that will be visited often, the admin notice is dismissable anyway... and lastly, we want to be as loud as possible regarding this change. 🙂